### PR TITLE
Improve warning about smart contract best practices

### PR DESCRIPTION
**Pull request by yosyp**
Fixed small typo, "also" should be "all".

### CODE REVIEW

**Comment by chriseth** [comment]
This is not a typo. The `also` was meant to state that since smart contracts are a special case of software, any guidelines for writing software "also" applies to smart contracts.

How did you interpret it?

**Comment by yosyp** [comment]
I interpreted the `also` as a mistake. Here are a few suggestions that me that sentence pair less awkward to read:
- _... Therefore, even smart contract should be ..._
- _... This software is no exception, and all smart contracts should be created following ..._
- _All software is written by humans, and may contain bugs. Since smart contracts are a special case of software, any guidelines for writing software also apply to smart contracts, and should be created following well-known best-practices in software development._

**Comment by chriseth** [comment]
My personal preference would be the last version. Thanks a lot for looking into this! Unfortunately, we do not have a native English speaker on our team :)

**Comment by yosyp** [comment]
Not a problem. Will be glad to read over and correct documentation as I learn more about the project.

Should I make a new PR or can the approving reviewer modify my commit?

**Comment by chriseth** [comment]
If you force-push to the same branch, the pull request will just update, so please do not open a new one.

**Review by axic — APPROVED** [review]

**Comment by axic** [comment]
Force pushed myself.

@chriseth I took the last suggested text, merge if you approve

File: docs/index.rst [hunk 1]
```diff
@@ -24,9 +24,11 @@ crowdfunding, blind auctions, multi-signature wallets and more.
     and run the smart contracts. 
 
 .. warning::
-    Since software is written by humans, it can have bugs. Thus, also
-    smart contracts should be created following well-known best-practices in
-    software development. This includes code review, testing, audits and correctness proofs.
+    Since software is written by humans, and may contain bugs.
```
Comment (chriseth) [inline docs/index.rst:27]: `and may` does not sound right to me.

File: docs/index.rst [hunk 2]
```diff
@@ -24,9 +24,11 @@ crowdfunding, blind auctions, multi-signature wallets and more.
...
+    Since smart contracts are a special case of software,
```
Comment (chriseth) [inline docs/index.rst:28]: The previous sentence already started with `since`.

File: docs/index.rst [hunk 3]
```diff
@@ -24,9 +24,11 @@ crowdfunding, blind auctions, multi-signature wallets and more.
...
+    any guidelines for writing software also apply to smart contracts,
+    and should be created following well-known best-practices in software development.
```
Comment (chriseth) [inline docs/index.rst:30]: `and should be created` sounds like the guidelines should be created in the way stated.

**Review by chriseth — CHANGES_REQUESTED** [review]
This looks like something went wrong in the rebase process...

**Comment by axic** [comment]
> This looks like something went wrong in the rebase process...

As mentioned "I took the suggested text", which was the 3rd from the list above.

**Comment by chriseth** [comment]
Oh I'm sorry, then I did not read the list properly. While I think the current paragraph should be expanded, I would not consider this PR an improvement at this stage.

File: docs/index.rst (hunk 1 above)
Comment (axic) [reply docs/index.rst:27]: Argh, it was a bad copy paste job, it should read `All software` from the original comment. Not that it is much better :)

**Comment by chriseth** [comment]
The first sentence is still weird and as there is no progress on this, I'm closing this PR.

**Pull request closed without merging**